### PR TITLE
Improve photo album pages

### PR DESCRIPTION
**Pull request by eikhr**
# Description

The image gallery was a bit clunky and had some bugs.

The following issues are fixed in this PR:
- [x] Opening and closing an image in an album resets scroll position
- [x] Photo galleries used an old, unnecessary `ProgressiveImage` component
- [x] Going to next and previous image in single photo view in album is sloooow
- [x] Photos have different heights, making it hard to hit the next/previous buttons multiple times
- [x] When closing a modal with the button, the element behind the close button is also pressed
- [x] Photo gallery modal is not dismissable with [esc] or by pressing outside it

Should be tested to really feel the improvements.

# Result

The above issues are fixed. Minor visual changes.

- [x] Changes look good on both light and dark theme.
- [x] Changes look good with different viewports (mobile, tablet, etc.).
- [x] Changes look good with slower Internet connections.

<table>
    <tr>
        <td width="25%">Description</td>
        <td>Before</td>
        <td>After</td>
    </tr>
    <tr>
        <td>
            Photo modal on mobile
        </td>
        <td>

webapp-staging abakus no_photos_423_picture_30553(iPhone 12 Pro)
        </td>
        <td>

localhost_3000_photos_427(iPhone 12 Pro)
        </td>
</tr>
<tr>
        <td>
            Photo edit modal (could use some more work...) <br/> It now has a lot less duplicated code, though.
        </td>
        <td>

webapp-staging abakus no_photos_423_picture_30553(iPhone 12 Pro) (2)
        </td>
        <td>

localhost_3000_photos_427(iPhone 12 Pro) (1)
        </td>
    </tr>
</table>

# Testing

- [x] I have thoroughly tested my changes.

Please describe what and how the changes have been tested, and provide instructions to reproduce if necessary.

---

Resolves ABA-1159

### CODE REVIEW

File: app/reducers/galleryPictures.ts [hunk 1]
```diff
@@ -23,7 +23,11 @@ export const initialUploadStatus: UploadStatus = {
   showStatus: false,
 };
 
-const legoAdapter = createLegoAdapter(EntityType.GalleryPictures);
+const legoAdapter = createLegoAdapter(EntityType.GalleryPictures, {
+  sortComparer: (a, b) => {
+    return Number(a.id) - Number(b.id);
```
Comment (jonasdeluna) [inline app/reducers/galleryPictures.ts:28]: Implicit return?

**Review by jonasdeluna — APPROVED** [review]
Awesome! If it works it works 👍

File: app/routes/photos/components/GalleryPictureEditForm.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            <Button
+              flat
+              onPress={() => {
+                navigate(`/photos/${gallery.id}/picture/${picture.id}`);
+              }}
+            >
+              Avbryt
+            </Button>
+            <SubmitButton>Lagre</SubmitButton>
+          </ButtonGroup>
```
Comment (ivarnakken) [inline app/routes/photos/components/GalleryPictureEditForm.tsx:108]: Why remove the "Slett bildet" button?

File: app/components/ProgressiveImage/index.tsx
Comment (ivarnakken) [inline app/components/ProgressiveImage/index.tsx:1]: Wait is this not a thing anymore? Is it handled by the Image component directly?

File: app/components/HeaderNotifications/index.tsx [hunk 3]
```diff
@@ -118,7 +118,7 @@ const NotificationsDropdown = () => {
       }}
       closeOnContentClick
       triggerComponent={
-        <Icon.Badge
+        <BadgeIcon
```
Comment (ivarnakken) [inline app/components/HeaderNotifications/index.tsx:121]: Why is this new

File: app/actions/GalleryPictureActions.ts [hunk 4]
```diff
@@ -32,28 +32,6 @@ export const fetchGalleryPictures = (
...
-      page_size: 1,
-      cursor,
-    },
-    schema: [galleryPictureSchema],
-    meta: {
-      errorMessage: 'Henting av bilde feilet',
-    },
-    propagateError: true,
-  });
-}
```
Comment (ivarnakken) [inline app/actions/GalleryPictureActions.ts:55]: Is this not useful anymore? I suppose it's nice to fetch the siblings when the current image is finished loading?

File: app/routes/photos/components/GalleryPictureModal.module.css [hunk 5]
```diff
@@ -1,14 +1,27 @@
 @import url('~app/styles/variables.css');
 
+.loadingIndicator {
+  margin: 0.375rem;
```
Comment (ivarnakken) [inline app/routes/photos/components/GalleryPictureModal.module.css:4]: no

File: app/routes/photos/components/GalleryPictureModal.module.css [hunk 6]
```diff
@@ -22,6 +35,9 @@
 }
 
 .dropdown {
+  position: absolute;
+  top: 0;
+  right: 3rem;
```
Comment (ivarnakken) [inline app/routes/photos/components/GalleryPictureModal.module.css:40]: no

File: app/routes/photos/components/GalleryPictureModal.tsx [hunk 7]
```diff
@@ -376,22 +399,31 @@ const GalleryPictureModal = () => {
...
+          >
+            <Icon
+              onPress={previousGalleryPicture}
+              name="arrow-back-outline"
+              size={40}
+              disabled={isFirstImage}
+            />
+            {isLastLoadedImage && !isLastImage ? (
+              <LoadingIndicator loading className={styles.loadingIndicator} />
+            ) : (
```
Comment (ivarnakken) [inline app/routes/photos/components/GalleryPictureModal.tsx:415]: Skeleton component? 😢

**Review by ivarnakken — APPROVED** [review]
This looks much better! 😍 I like

**Comment by ShaileshS1702** [comment]
It look so much nicer!

File: app/routes/photos/components/GalleryPictureEditForm.tsx (hunk 2 above)
Comment (eikhr) [reply app/routes/photos/components/GalleryPictureEditForm.tsx:108]: It is also in the dropdown. Why select "Edit" -> "Delete", when you could just do "Delete" directly

File: app/components/ProgressiveImage/index.tsx
Comment (eikhr) [reply app/components/ProgressiveImage/index.tsx:1]: Yeh, I'm not sure exactly what this did. But the Image-component supports progressive loading, so I just used theat instead

File: app/components/HeaderNotifications/index.tsx (hunk 3 above)
Comment (eikhr) [reply app/components/HeaderNotifications/index.tsx:121]: `Icon.Badge` wasn't possible anymore with the `forwardRef` in `Icon`.

File: app/actions/GalleryPictureActions.ts (hunk 4 above)
Comment (eikhr) [reply app/actions/GalleryPictureActions.ts:55]: Nah, it doesn't really make any sense as this just fetches one image at a time. We can just fetch a whole page of images at a time, like in the album-page. This just causes unneccessary requests.

File: app/routes/photos/components/GalleryPictureModal.module.css (hunk 5 above)
Comment (eikhr) [reply app/routes/photos/components/GalleryPictureModal.module.css:4]: It should match the padding of clickable `Icon`s: https://github.com/webkom/lego-webapp/blame/8d156d0615da926ca9b2d0cc12037576fa4bc223/packages/lego-bricks/src/components/Icon/Icon.module.css#L24

File: app/routes/photos/components/GalleryPictureModal.tsx (hunk 7 above)
Comment (eikhr) [reply app/routes/photos/components/GalleryPictureModal.tsx:415]: This is on the next button. The current image is still displayed as normal:)
Looks like this while loading: 
Screenshot 2024-11-05 at 17 15 17

File: app/routes/photos/components/GalleryPictureModal.module.css (hunk 5 above)
Comment (eikhr) [reply app/routes/photos/components/GalleryPictureModal.module.css:4]: It avoids a slight movement in the  layout, which doesn't look great.

**Pull request merged**